src/renderers/shared/server/ReactPartialRenderer.js

```diff
@@ -827,6 +830,7 @@ class ReactDOMServerRenderer {
       frame.debugElementStack = [];
     }
     this.stack.push(frame);
+    this.previousWasTextNode = false;
```

Comment:
So is this unnecessary now? Seems like the fix was the second commit, not this, no?

Reply:
If I remember it right I tried commenting both places out separately after adding the tests, and both were necessary. This one resets when we're entering a tag, and another resets when we're leaving a tag.